homeassistant/components/cover/insteon.py

```diff
@@ -0,0 +1,77 @@
+"""
+Support for Insteon lights via PowerLinc Modem.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/light.insteon/
+"""
+import asyncio
+import logging
+import math
+
+from homeassistant.components.insteon import InsteonEntity
+from homeassistant.components.cover import (CoverDevice, ATTR_POSITION,
+                                            SUPPORT_OPEN, SUPPORT_CLOSE,
+                                            SUPPORT_SET_POSITION)
+# from homeassistant.const import (
+#     STATE_OPEN, STATE_CLOSED, STATE_OPENING, STATE_CLOSING, STATE_UNKNOWN)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['insteon']
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_entities,
+                         discovery_info=None):
+    """Set up the Insteon component."""
+    insteon_modem = hass.data['insteon'].get('modem')
```

Comment:
If the platform is only set up via discovery from the component, we should add a guard clause here that checks of discovery_info is None and returns.

Reply:
Under what condition would discovery_info be None? If this platform is called and exit immediately because dicovery_info is None, I would want to post a log message but not sure what that message should be.

Reply:
Also, I assume it should just return, rather than return False.

Reply:
We usually don't bother with a log message in that case. It would happen if a user by mistake adds a config section for the platform under the cover component. But if that isn't supported and not documented, a log message might be more confusing than helpful.

Reply:
Ignore the questions. I found other code to model in appletv.py.